torch/csrc/jit/mobile/import.cpp

```diff
@@ -520,10 +520,32 @@ c10::IValue BytecodeDeserializer::readArchive(
       return obj;
     }
   };
-
```

Comment:
What's the main difference in logic between the 2 imports?

This is related to my general question about having a single place that handles the versioning transformation logic.

Reply:
The main difference in logic is that in v4, we have "17" indicating the tensor is saved in the folder with the same archive name. For example, 

Before:
in bytecode.pkl, we have a tensor metadata as 
```
(tensor 17, dims, ..., "17", ...)
(tensor 18, dims, ..., "18", ...)
```
The tensor binary is saved in file
bytecode/17
bytecode/18
...

After: 
We reuse the files from constants, and only save tensors that are not in constants. So the format of bytecode.pkl changes to: 
```
(tensor 17, dims, ..., "constants/17", ...)
(tensor 18, dims, ..., "bytecode/0", ...)
```

Reply:
I think @raziel  meant the difference between `torch/csrc/jit/mobile/import.cpp` and `torch/csrc/jit/serialization/import.cpp`. `torch/csrc/jit/mobile/import.cpp` is for importing bytecode, and `torch/csrc/jit/serialization/import.cpp` is for importing jit, like @kimishpatel mentioned.